Add OrchardCore.ContentField.Core project

CrestApps opened the pull request:
This pull request Fix #11678 

Do we need to move the `setting` classes along with the view-models to this project too? Or should we just leave them where there are at? Perhaps, we can place the fields into `.Abstraction` project where the fields would live while settings and viewmodel would live in `.Core` project.

sebastienros commented:
I assume this wouldn't be a breaking change since the module would still reference the Core module?

CrestApps commented:
@sebastienros yes this should not break anything since all namespaces are the same the the OrchardCore.ContentFields still depends on projects that has all the dependencies.

@Skrypt I just made a change where I moved all the field and settings models to a new `.Abstractions` project and moved services to the `.Core` project. At this point, I am not sure if we need `.Core` project but may not be a bad idea to keep it. With this change any user that wants to use fields in their module, they would only install `OrchardCore.ContentFields.Abstractions` this would provide the content fields model and the settings needed.

hishamco commented:
Please fix the conflict

hishamco commented on src/OrchardCore.Modules/OrchardCore.ContentFields/OrchardCore.ContentFields.csproj:
```diff
@@ -16,6 +18,8 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\OrchardCore\OrchardCore.Admin.Abstractions\OrchardCore.Admin.Abstractions.csproj" />
+    <ProjectReference Include="..\..\OrchardCore\OrchardCore.ContentFields.Abstractions\OrchardCore.ContentFields.Abstractions.csproj" />
```
We need to reference the `OC.ContentFields.Core` ONLY

hishamco commented on src/OrchardCore/OrchardCore.ContentFields.Abstractions/OrchardCore.ContentFields.Abstractions.csproj:
```diff
@@ -0,0 +1,24 @@
+    <Description>
+      $(OCCMSDescription)
+
+      Abstractions for ContentFields Module.
```
Abstractions for ContentFields

Skrypt commented:
Please fix conflicts by merging latest main branch.

CrestApps commented:
@Skrypt I am not sure how to do this one. If this is something we are planing on merging I can just create a new PR for this.

![image](https://user-images.githubusercontent.com/24724371/178124552-55e8d219-1f00-419e-8efb-6884332ffa3c.png)

hishamco commented:
No need to create another PR, try to rebase the `main` branch

CrestApps commented:
@Skrypt 

Closing this PR in favor of PR #11990.

CrestApps commented:
@hishamco I tried but could not pull it off. I already pushed a cleaner PR as a replacement to this.
